packages/shared/src/components/notifications/EnableNotification.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { ReactElement, useContext, useEffect } from 'react';
+import classNames from 'classnames';
+import usePersistentContext from '../../hooks/usePersistentContext';
+import { Button } from '../buttons/Button';
+import NotificationsContext from '../../contexts/NotificationsContext';
+import CloseButton from '../CloseButton';
+import { cloudinary } from '../../lib/image';
+import VIcon from '../icons/V';
+import { webappUrl } from '../../lib/constants';
+import { useAnalyticsContext } from '../../contexts/AnalyticsContext';
+import { AnalyticsEvent } from '../../lib/analytics';
+
+export enum NotificationPromptSource {
+  NotificationsPage = 'notifications page',
+  NewComment = 'new comment',
+  CommunityPicks = 'community picks modal',
+}
+
+const DISMISS_PERMISSION_BANNER = 'DISMISS_PERMISSION_BANNER';
+
+type EnableNotificationProps = {
+  source?: NotificationPromptSource;
+  parentCommentAuthorName?: string;
+};
+
+const containerClassName: Record<NotificationPromptSource, string> = {
+  [NotificationPromptSource.NotificationsPage]:
+    'px-6 w-full border-l bg-theme-float',
+  [NotificationPromptSource.NewComment]: 'rounded-16 border px-4 mt-3',
+  [NotificationPromptSource.CommunityPicks]: 'rounded-16 border px-4 mt-3',
+};
+
+function EnableNotification({
+  source = NotificationPromptSource.NotificationsPage,
+  parentCommentAuthorName,
+}: EnableNotificationProps): ReactElement {
+  const { trackEvent } = useAnalyticsContext();
+  const {
+    isSubscribed,
+    isNotificationSupported,
+    hasPermissionCache,
+    acceptedPermissionJustNow: isEnabled,
+    onTogglePermission,
+    onAcceptedPermissionJustNow,
+  } = useContext(NotificationsContext);
+  const [isDismissed, setIsDismissed, isLoaded] = usePersistentContext(
+    DISMISS_PERMISSION_BANNER,
+    false,
+  );
+  const onDismiss = () => {
+    trackEvent({
+      event_name: AnalyticsEvent.ClickNotificationDismiss,
+      extra: JSON.stringify({ origin: source }),
+    });
+    setIsDismissed(true);
+  };
+
+  const onEnable = async () => {
+    const permission = await onTogglePermission();
+
+    trackEvent({
+      event_name: AnalyticsEvent.ClickEnableNotification,
+      extra: JSON.stringify({ origin: source, permission }),
+    });
```

Comment:
This is now duplicate can we mabye abstract it to one implementation somewhere?

Reply:
This is a banner somewhere else. The place this is ideal to be abstracted is at `onTogglePermission` but there are other areas where it is used and has its own tracking events. This was separated intentionally to avoid duplication.

Reply:
Hmm and what if we abstract this tracking part? 
Just to avoid clutter and confusion down the line?

I think we do it for some other events already.

Reply:
Sure thing. Let me create a generic function for this 🚢